pkg/odo/cli/component/create.go

```diff
@@ -919,6 +921,14 @@ func downloadStarterProject(devObj parser.DevfileObj, projectPassed string, inte
 		return err
 	}
 
+	if co.devfileMetadata.starterToken == "" && registryUtil.IsSecure(co.devfileMetadata.devfileRegistry.Name) {
+		token, err := keyring.Get(fmt.Sprintf("%s%s", util.CredentialPrefix, co.devfileMetadata.devfileRegistry.Name), registryUtil.RegistryUser)
+		if err != nil {
+			return errors.Wrap(err, "unable to get secure registry credential from keyring")
+		}
+		co.devfileMetadata.starterToken = token
+	}
```

Comment:
little confused here, how we would know that starterProject or registry hosted by same user?

Reply:
possibly we can host a secure starter project here https://github.com/odo-devfiles for test purpose

Reply:
Currently we always use this const `registryUtil.RegistryUser` to represent the "default" user because odo doesn't support multiple users (Note: the definition of user here is the OS level user and we always use the same user to access the OS keychain to store/retrieve credentials for starter project or registry)

Reply:
@GeekArthur maybe I have not understand it clearly, but till have a doubt
i mean, how we are handling the case where registry and secure project both are hosted on different github repositories by different user, so it would have two different tokens for starter project and registry, here from code it seems we are directly using registry token.

Reply:
You are right. If the registry and starter project are hosted by different GitHub users, currently the user has to pass in the starter project token explicitly by using --starter --starter-token <starter project token> when creating the component as by default the starter project will use the secure registry token if the registry is secure. The reason to do this is currently GitHub only support user-scoped token for all repos, so if user hosts both registry and starter projects under same user/account then it would be easier for user to create component as user doesn't need to pass in starter project token every time.